libcontainer/intelrdt/intelrdt.go

```diff
@@ -495,8 +448,14 @@ func IsMBAEnabled() bool {
 
 // Check if Intel RDT/MBA Software Controller is enabled
 func IsMBAScEnabled() bool {
-	featuresInit()
-	return mbaScEnabled
+	// Neither libcontainer nor runc need to check, and parsing mountinfo is
+	// the only way to check it. Parse it lazily.
+	mi, err := findResctrlMountinfo()
+	if err != nil {
+		return false
+	}
+	// Check if MBA Software Controller is enabled through mount option "-o mba_MBps"
+	return strings.Contains(","+mi.VFSOptions+",", ",mba_MBps,")
```

Comment:
Had to look at this one a few times, as it looked a bit brittle at a glance, but `VFSOptions` comes from mountinfo, so I assume no whitespaces around `,` would exist here

^^ this whole comment would become irrelevant if we decide to remove this function altogether.

Reply:
First: I didn't write this line. I just moved it around.

I had to look at this one a few times as well. It has nothing to do with whitespace. It's a simple and elegant solution for testing if a comma-separated list contains a value without the false positives which would come with a naive substring match. The list is prepended and appended with commas to eliminate the edge cases of the first and last list items.

Reply:
> The list is prepended and appended with commas to eliminate the edge cases of the first and last list items.

Yup, got that one; was mostly wondering at first how strict `VFSOptions` formatting itself was, i.e. if

`","+mi.VFSOptions+","` could be `","+"some, options, mba_MBps , other_option"+","`, so if it could have whitespace in the middle.

Reply:
Ah. The list of options is synthesized by the kernel](https://github.com/torvalds/linux/blob/efb52a7d9511df818391f1afa459507425833438/arch/x86/kernel/cpu/resctrl/rdtgroup.c#L3184-L3196) when the mountinfo file is opened so no whitespace is possible

Reply:
Ah! Thx.

In either case, if we don't need this function, we don't have to worry about it at all 😄 

@kolyshkin @marquiz any reason we'd want to keep it? It looks unused, and would save a lot of extra complexity that's not needed if it's gone.

Reply:
`IsMBAScEnabled` first appeared in commit 95af9eff82157317817d3d550626f3e5db240d8a (as `IsMbaScEnabled`) and, despite being a public function, was only used by intelrdt unit tests. This usage was dropped by commit 1b4c30fd8b719834a5492a17893994a996c0a7ce, and since then this function is unused.

Yes, @thaJeztah is right, it should be removed in the first commit, simplifying further changes.

@corhere please 🙏🏻

Reply:
@thaJeztah @kolyshkin will do 👍